contracts/shapella-upgrade/ShapellaUpgradeTemplate.sol

```diff
@@ -0,0 +1,870 @@
+    uint8 status;
+    /// @notice name of staking module
+    string name;
+    /// @notice block.timestamp of the last deposit of the staking module
+    /// @dev NB: lastDepositAt gets updated even if the deposit value was 0 and no actual deposit happened
+    uint64 lastDepositAt;
+    /// @notice block.number of the last deposit of the staking module
+    /// @dev NB: lastDepositBlock gets updated even if the deposit value was 0 and no actual deposit happened
+    uint256 lastDepositBlock;
+    /// @notice number of exited validators
+    uint256 exitedValidatorsCount;
+}
+
+/**
+* @title Shapella Lido Upgrade Template
+*
+* @dev Auxiliary contracts which performs binding of already deployed Shapella upgrade contracts.
+* Must be used by means of two calls:
+*   - `startUpgrade()` before updating implementation of Aragon apps
+*   - `finishUpgrade()` after updating implementation of Aragon apps
+* The required initial on-chain state is checked in `startUpgrade()`
+*/
+contract ShapellaUpgradeTemplate {
+
+    bytes32 public constant DEFAULT_ADMIN_ROLE = 0x00;
+    uint256 public constant NOT_INITIALIZED_CONTRACT_VERSION = 0;
+
+    uint256 public constant _accountingOracleConsensusVersion = 1;
+    uint256 public constant _validatorsExitBusOracleConsensusVersion = 1;
+    string public constant NOR_STAKING_MODULE_NAME = "curated-onchain-v1";
+    bytes32 public constant _nodeOperatorsRegistryStakingModuleType = bytes32("curated-onchain-v1");
+    uint256 public constant _nodeOperatorsRegistryStuckPenaltyDelay = 172800;
+    bytes32 public constant _withdrawalCredentials = 0x010000000000000000000000dc62f9e8c34be08501cdef4ebde0a280f576d762;
+    uint256 public constant NOR_STAKING_MODULE_ID = 1;
+    uint256 public constant NOR_STAKING_MODULE_TARGET_SHARE_BP = 10000; // 100%
+    uint256 public constant NOR_STAKING_MODULE_MODULE_FEE_BP = 500; // 5%
+    uint256 public constant NOR_STAKING_MODULE_TREASURY_FEE_BP = 500; // 5%
+    uint256 public constant VEBO_LAST_PROCESSING_REF_SLOT = 0;
+
+    ILidoLocator public constant _locator = ILidoLocator(0x1eDf09b5023DC86737b59dE68a8130De878984f5);
+    IHashConsensus public constant _hashConsensusForAccountingOracle = IHashConsensus(0x8d87A8BCF8d4e542fd396D1c50223301c164417b);
+    IHashConsensus public constant _hashConsensusForValidatorsExitBusOracle = IHashConsensus(0x8374B4aC337D7e367Ea1eF54bB29880C3f036A51);
+    address public constant _eip712StETH = 0xB4300103FfD326f77FfB3CA54248099Fb29C3b9e;
+    address public constant _voting = 0xbc0B67b4553f4CF52a913DE9A6eD0057E2E758Db;
+    address public constant _agent = 0x4333218072D5d7008546737786663c38B4D561A4;
+    INodeOperatorsRegistry public constant _nodeOperatorsRegistry = INodeOperatorsRegistry(0x9D4AF1Ee19Dad8857db3a45B0374c81c8A1C6320);
+    address public constant _gateSeal = 0x75A77AE52d88999D0b12C6e5fABB1C1ef7E92638;
+    address public constant _withdrawalQueueImplementation = 0xF7a378BB9E911550baA5e729f5Ab1592aDD905A5;
```

Comment:
Let's hardcode implementations for `Lido`, `LidoOracle`, and `NOR` before the mainnet deployment:
- can add more asserts for the post-deployment state
- can ask auditors for the whole address set verification

Reply:
done